### PR TITLE
For #10921 - Integrate BrowserThumbnails feature into the tabs tray

### LINKED ISSUE #10921: FNX-4690 ⁃ Integrate BrowserThumbnails into the tabs tray
When https://github.com/mozilla-mobile/android-components/issues/7021 lands, we should look at integrating the BrowserThumbnails into the tabs tray.

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #10921 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt [hunk 1]
```diff
@@ -63,8 +66,16 @@ class TabTrayViewHolder(
...
-        iconView?.setImageBitmap(tab.icon)
+
+        if (tab.thumbnail != null) {
+            thumbnailView.setImageBitmap(tab.thumbnail)
+            thumbnailView.visibility = View.VISIBLE
+            iconView?.visibility = View.INVISIBLE
+        } else {
+            thumbnailView.visibility = View.INVISIBLE
+            iconView?.setImageBitmap(tab.icon)
+            iconView?.visibility = View.VISIBLE
```
Comment (darkwing) [inline app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt:77]: Lame nit but could we get these in the same order in both sides of the if/else?

**Comment by darkwing** [comment]
I tested briefly by:

1.  Going to https://davidwalsh.name -- yay, got a screenshot!  Excellent!
2.  Opened a second tab, https://espn.com -- crash

I don't see crash detail in my logs, but I do see a "read: unexpected EOF!"

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt (hunk 1 above)
Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt:77]: Fixed.

**Comment by gabrielluong** [comment]
> I tested briefly by:
> 
> 1. Going to https://davidwalsh.name -- yay, got a screenshot!  Excellent!
> 2. Opened a second tab, https://espn.com -- crash
> 
> I don't see crash detail in my logs, but I do see a "read: unexpected EOF!"

I couldn't reproduce the issue. If you see it again, could you post the full logs?

In the meantime, I am gonna go ahead with landing this so we can get more broad testing.

**Pull request merged**